packages/open-truss/src/configuration/engine-v1/config-schemas.tsx

```diff
@@ -95,32 +95,33 @@ export type WorkflowV1 = z.infer<typeof WorkflowV1Shape>
 
 export const BaseOpenTrussComponentV1PropsShape = z.object({
   data: DataV1Shape,
-  config: WorkflowV1Shape,
+  config: WorkflowV1Shape.optional(),
 })
-export const withChildren = (shape: z.AnyZodObject): z.AnyZodObject =>
-  shape.extend({ children: z.any().optional() })
-const ComponentWithChildrenShape = withChildren(
-  BaseOpenTrussComponentV1PropsShape,
-)
-type ComponentWithChildren = z.infer<typeof ComponentWithChildrenShape>
 
-export type BaseOpenTrussComponentV1Props =
-  | z.infer<typeof BaseOpenTrussComponentV1PropsShape>
-  | ComponentWithChildren
+type BaseOpenTrussComponentV1Props = z.infer<
+  typeof BaseOpenTrussComponentV1PropsShape
+>
 
-export type BaseOpenTrussComponentV1 = (
-  props: BaseOpenTrussComponentV1Props,
+export type BaseOpenTrussComponentV1<
+  AdditionalProps = Record<string, unknown>,
+> = (
+  props: BaseOpenTrussComponentV1Props & AdditionalProps,
 ) => React.JSX.Element
 
-const FrameWrapperShape = withChildren(
-  BaseOpenTrussComponentV1PropsShape,
-).extend({
+export const withChildren = { children: z.any().optional() }
+const ComponentWithChildrenShape =
+  BaseOpenTrussComponentV1PropsShape.extend(withChildren)
+type ComponentWithChildren = z.infer<typeof ComponentWithChildrenShape>
+
+const FrameWrapperShape = BaseOpenTrussComponentV1PropsShape.extend({
```

Comment:
`BaseOpenTrussComponentV1PropsShape`'s properties are already included when you use the `BaseOpenTrussComponentV1<>` generic on line 122. Do you need to extend this shape or can you just use `z.object`?

Reply:
You're correct, this is duplicative. I did this because it's how OT-compatible components do it and `FrameWrapper` is one so I did the same thing. I hadn't thought of just doing `object` but I think it would be equivalent. I think I like this way better because it keeps things aligned and we do want regular OT components to do this by extending the base shape (we `export const Props` those ones and if it were just an extension of `object` it would miss the base props). What do you think?

Reply:
That makes sense to me!